apps/src/lib/levelbuilder/code-docs-editor/ProgrammingEnvironmentEditor.jsx

```diff
@@ -18,7 +20,35 @@ const useProgrammingEnvironment = initialProgrammingEnvironment => {
     setProgrammingEnvironment({...programmingEnvironment, [key]: value});
   };
 
-  return [programmingEnvironment, updateProgrammingEnvironment];
+  return [
+    programmingEnvironment,
+    updateProgrammingEnvironment,
+    setProgrammingEnvironment
+  ];
+};
+
+const renderCategoryEditor = (category, updateFunc) => {
+  return (
+    <div>
+      <label>
+        Name
+        <input
+          value={category.name || ''}
+          onChange={e => updateFunc('name', e.target.value)}
+          style={styles.textInput}
+        />
+      </label>
+      <label>
+        Color
+        <input
```

Comment:
How does this input turn into the cool color picker?

Reply:
`type="color"` below does this, the same way `type="checkbox"` turns an input into a checkbox.

Reply:
TIL!